lib/widgets.php

```diff
@@ -100,33 +105,38 @@ function gutenberg_get_legacy_widget_settings() {
 
 	$has_permissions_to_manage_widgets = current_user_can( 'edit_theme_options' );
 	$available_legacy_widgets          = array();
-	global $wp_widget_factory, $wp_registered_widgets;
-	foreach ( $wp_widget_factory->widgets as $class => $widget_obj ) {
-		$available_legacy_widgets[ $class ] = array(
-			'name'             => html_entity_decode( $widget_obj->name ),
-			// wp_widget_description is not being used because its input parameter is a Widget Id.
-			// Widgets id's reference to a specific widget instance.
-			// Here we are iterating on all the available widget classes even if no widget instance exists for them.
-			'description'      => isset( $widget_obj->widget_options['description'] ) ?
-				html_entity_decode( $widget_obj->widget_options['description'] ) :
-				null,
-			'isCallbackWidget' => false,
-			'isHidden'         => in_array( $class, $core_widgets, true ),
-		);
+	if ( ! empty( $GLOBALS['wp_widget_factory'] ) ) {
```

Comment:
Minor: Would calling `global $wp_widget_factory;` when the global isn't assigned be problematic, and then proceed to check `! empty( $wp_widget_factory )` ? Is only a minor difference to avoid reference to the superglobal.

Reply:
In my tests that approach also seems to work well, the code was updated 👍